File: Excel/800-899/808/808 Challenge.py
import pandas as pd

path = "800-899/808/808 Group By Fruits.xlsx"
input = pd.read_excel(path, usecols="A", skiprows=1, nrows=16, names=["Data"])
test = pd.read_excel(path, usecols="C:D", skiprows=1, nrows=6)

input = input.dropna()['Data'].str.split(', ', expand=True).stack().reset_index(drop=True)
fruits = input[~input.str.isdigit()].reset_index(drop=True)
weights = input[input.str.isdigit()].astype(int).reset_index(drop=True)
result = pd.DataFrame({'Fruits': fruits, 'Weight': weights})
result = result.groupby('Fruits', as_index=False)['Weight'].sum().rename(columns={'Weight': 'Total Weight'}).sort_values('Fruits').reset_index(drop=True)

print(result.equals(test)) # True

File: Excel/800-899/808/808 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/800-899/808/808 Group By Fruits.xlsx"
input = read_excel(path, range = "A2:A18")
test  = read_excel(path, range = "C2:D8")

result = input %>%
  separate_longer_delim(Data, delim = ", ") %>%
  mutate(measure = ifelse(str_detect(Data, "^[0-9]+$"), "Weight", "Fruits")) %>%
  mutate(rn = row_number(), .by = measure) %>%
  pivot_wider(names_from = measure, values_from = Data) %>%
  summarise(`Total Weight` = sum(as.numeric(Weight)), .by = Fruits) %>%
  arrange(Fruits)

all.equal(result, test)
# TRUE